airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/formats/csv_parser.py

```diff
@@ -239,12 +239,18 @@ def stream_records(self, file: Union[TextIO, BinaryIO], file_info: FileInfo) ->
         schema = self._get_schema_dict_without_inference(file)
         schema.update(self._master_schema)
 
-        streaming_reader = pa_csv.open_csv(
-            file,
-            pa.csv.ReadOptions(**self._read_options()),
-            pa.csv.ParseOptions(**self._parse_options()),
-            pa.csv.ConvertOptions(**self._convert_options(schema)),
-        )
+        try:
+            streaming_reader = pa_csv.open_csv(
+                file,
+                pa.csv.ReadOptions(**self._read_options()),
+                pa.csv.ParseOptions(**self._parse_options()),
+                pa.csv.ConvertOptions(**self._convert_options(schema)),
+            )
+        except ArrowInvalid as e:
+            if "try to increase block size?" in str(e):
+                error_message = "Unable to determine two block bounderies. Try to change Block Size value"
+                raise AirbyteTracedException(message=error_message, failure_type=FailureType.config_error) from e
+            raise AirbyteTracedException(message=str(e), failure_type=FailureType.config_error) from e
```

Comment:
```suggestion
            error_message = str(e)
            if "try to increase block size?" in error_message:
                error_message = "Unable to determine two block boundaries. Try to change Block Size value."
            raise AirbyteTracedException(message=str(e), failure_type=FailureType.config_error) from e
```

Reply:
What would be a benefit of your version?

Reply:
Those repetitive raises do not violate DRY principle in a sence that it does not force us to the edit code in multiple places in the future. It does not matter whether we edit the `error_message` as an `AirbyteTracedException` argument or just a variable: it will be done at the same place only once. So could you please help me understand how the DRY principle can even be applied to this situation? Initially I wanted the code to look like this:
```
if <condition>:
    raise AirbyteTracedException(message="Unable to determine two block bounderies. Try to change Block Size value", failure_type=FailureType.config_error) from e
```
but this code row was too long and that was only the reason why I made the `error_message` variable on top of the `raise` expression. If I had known that it will lead to such a missunderstanding I would have constructed it like the following:

```
if <condition>:
    raise AirbyteTracedException(
        message="Unable to determine two block bounderies. Try to change Block Size value", 
        failure_type=FailureType.config_error
    ) from e
```
but at that moment I thought it is a little bulky. But now I see it is not so bulky and much better then maintain that additional `error_message` variable.

So your code just seems a little bit more clean than mine and that's it. But in the future if we want to add more conditions then the more conditions we add the more ineffective your code will be. Because we will need to pass through all the conditions before finally raise an exception, when in my code we raise it once meet a condition.